tests/test_utils_datatypes.py

```diff
@@ -246,6 +247,19 @@ def test_cache_with_limit(self):
         self.assertEqual(cache['b'], 2)
         self.assertEqual(cache['c'], 3)
 
+    def test_cache_time_limit(self):
+        cache = LocalCache(time_limit=2)
+        cache['a'] = 1
+        cache['b'] = 2
+        cache['c'] = 3
+        self.assertIn('a', cache)
+        self.assertIn('b', cache)
+        self.assertIn('c', cache)
+        sleep(3)
+        self.assertNotIn('a', cache)
+        self.assertNotIn('b', cache)
+        self.assertNotIn('c', cache)
+
```

Comment:
Rather than sleeping, we should use something like https://github.com/spulec/freezegun to mock the pass of time.

I also don’t think having 3 entries helps much. I can understand testing more than 1 entry in case there’s ever an issue which causes only 1 entry (e.g. first or last) to be removed, but testing 2 entries should be enough to catch that kind of issue.

Finally, I think there’s additional scenarios worth testing:
- Test that entries are still there 1 second before they should be removed.
- Test that adding 1 entry, then another entry 1 second later, makes it so that after 1 second the first entry has been removed but the second entry remains, and is removed 1 second later.